### DISCUSSION TITLE
Implement it in functional Pathology data

### DISCUSSION THREAD

**Discussion by miladalipour99** [timeseriesAI/tsai#371] [Q&A] [answered]
Hello dear Oguiza 
I have a functional Biomedical data(Pathology) with shape of  [105 samples x 100 features x 130 timesteps]  
I aim to re develop the MVP for this kind of data and I need your kind advice for building the data for my own dataset.
Would you mind to showing me which notebooks will help me to redesign data without get_UCR_function as I dont use the UCR data?

**Comment by williamsdoug** [comment]
Hi @miladalipour99 .    It would be best to move this to Discussions it is not technically an issue.

I generally use Numpy arrays for as input for my work, although other tsai users often use Pandas Dataframes.  My suggestion is that tutorial_nbs/00b_How_to_use_numpy_arrays_in_fastai.ipynb as a good starting point.  If you look at the output of the 

```X_train, y_train, X_valid, y_valid = get_UCR_data(dsid, parent_dir='./data/UCR/', verbose=True, on_disk=False)``` 

statement, it simply a set of numpy arrays, where X_train/X_valid is the input timeseries data and y_train/y_valid are  data labels.  This is the format you'll want to match when loading your own data.

X_* are 3D arrays organized as [sample, features, timesteps], so you'll want to structure your data in a similar manner.

Before spending much time using tsai I  would recommend you go through the full set of tutorials -- I found them very helpful when I was starting to learn tsai.  You should be able to run these tutorials on colab, and I would suggest you add your own print statements to explore data formats or other areas that may be in question.  You may also be able to use some of these tutorals as a starting point for your work.

**Comment by oguiza** [comment, marked as answer]
Thanks, Doug!
